packages/backend/src/managers/catalogManager.ts

```diff
@@ -274,4 +286,68 @@ export class CatalogManager extends Publisher<ApplicationCatalog> implements Dis
   private getUserCatalogPath(): string {
     return path.resolve(this.appUserDirectory, USER_CATALOG);
   }
+
+  public filterRecipes(filters: RecipeFilters): FilterRecipesResult {
+    let result = this.getRecipes();
+    for (const [filter, values] of Object.entries(filters)) {
+      switch (filter) {
+        case 'languages': {
+          let res: Recipe[] = [];
+          for (const value of values) {
+            res = [...res, ...result.filter(r => r.languages?.includes(value))];
```

Comment:
why not result = result.filter ?

Reply:
To get an OR over all the values of a field: for example, if the user selects Javascript and Go in languages,  this will reutrn all the recipes having the lang Javascript OR the lang Go.